receiver/sqlserverreceiver/config.go

```diff
@@ -35,9 +34,15 @@ func (cfg *Config) Validate() error {
 
 	if !directDBConnectionEnabled(cfg) {
 		if cfg.Server != "" || cfg.Username != "" || string(cfg.Password) != "" {
-			return fmt.Errorf("Found one or more of the following configuration options set: [server, port, username, password]. " +
+			return fmt.Errorf("Found one or more of the following configuration options set: [server, username, password]. " +
```

Comment:
For some reason in my initial implementation the `port` option was checked in `directDBConnectionEnabled`. The result of this was currently (before this PR) no error is logged in `Validate` from a `port` being unspecified. However, since `0` is the default value and can't be used to connect to, the receiver would never successfully be able to scrape the given `server:port.`

This actually comes in handy now though, as with this proposal `port` itself is not required, but a port in either `server` or `port` **is** required. Now we will fail `Validate` if no port is specified (in either `server` or `port`).

This may be considered a bug fix so I can add a bug fix changelog to this as well, but the config that was passing validation when it should have failed wouldn't work anyways, so not sure if it needs a changelog.